pandas/core/internals/blocks.py

```diff
@@ -974,10 +974,19 @@ def setitem(self, indexer, value) -> Block:
                     # checking lib.is_scalar here fails on
```

Comment:
I can't comment on it, but a few lines above we have:

```python
        except LossySetitemError:
            # current dtype cannot store value, coerce to common dtype
            nb = self.coerce_to_target_dtype(value)
            return nb.setitem(indexer, value)
```

Is it theoretically possible that the dtype cannot store the value, but that the astype operation can still be done with a view? And should we in that case pass through `using_cow=using_cow` to the recursive `setitem` call?

I can't directly think of a case (maybe an EA that is backed by object array but can't hold the object you are setting?), but maybe the safest option is to just pass it.

Reply:
Hm I was thinking about it this as well, but I could not think of a case where we would get a LossySetitemError and then would not make a copy in the astype. Can add it if you prefer anyways